src/utils/async.js

```diff
@@ -1,5 +1,44 @@
 /* @flow strict-local */
 
+export class TimeoutError extends Error {
+  name = 'TimeoutError';
+}
+
+/**
+ * Time-out a Promise after `timeLimitMs` has passed.
+ *
+ * Returns a new Promise with the same outcome as `promise`, if
```

Comment:
Yeah; I looked around for a more common term to use; the [MDN doc on `Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) just uses "outcome" once.

I meant to write the interface without assuming `promise` will resolve; instead, it might reject. See Greg's comment at https://github.com/zulip/zulip-mobile/pull/4166#discussion_r442567648.

Maybe something like this:

```
 * Returns a new Promise with the same outcome (resolved/rejected) as
 * `promise`, if `promise` completes in time.
 *
 * If `promise` does not complete before `timeLimitMs` has passed,
 * `onTimeout` is called, and its outcome is used as the outcome of
 * the returned Promise.
```